### PR TITLE
8338156: Fix -Wzero-as-null-pointer-constant warnings in jvmciCompilerToVM.cpp

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this change to remove -Wzero-as-null-pointer-constant warnings
in jvmciCompilerToVM.cpp. We're changing uses of _0 suffixed exception macros
to instead use the corresponding _NULL suffixed macros where appropriate.

Testing: mach5 tier1

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8338156](https://bugs.openjdk.org/browse/JDK-8338156): Fix -Wzero-as-null-pointer-constant warnings in jvmciCompilerToVM.cpp (**Enhancement** - P4)

### Reviewers
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Julian Waters](https://openjdk.org/census#jwaters) (@TheShermanTanker - Committer)
 * [Doug Simon](https://openjdk.org/census#dnsimon) (@dougxc - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20538/head:pull/20538` \
`$ git checkout pull/20538`

Update a local copy of the PR: \
`$ git checkout pull/20538` \
`$ git pull https://git.openjdk.org/jdk.git pull/20538/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20538`

View PR using the GUI difftool: \
`$ git pr show -t 20538`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20538.diff">https://git.openjdk.org/jdk/pull/20538.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20538#issuecomment-2283265100)

### CODE REVIEW

File: src/hotspot/share/jvmci/jvmciCompilerToVM.cpp [hunk 1]
```diff
@@ -549,7 +549,7 @@ C2V_END
 C2V_VMENTRY_NULL(jobject, getImplementor, (JNIEnv* env, jobject, ARGUMENT_PAIR(klass)))
   Klass* klass = UNPACK_PAIR(Klass, klass);
   if (!klass->is_interface()) {
-    THROW_MSG_0(vmSymbols::java_lang_IllegalArgumentException(),
+    THROW_MSG_NULL(vmSymbols::java_lang_IllegalArgumentException(),
         err_msg("Expected interface type, got %s", klass->external_name()));
```
Comment (tschatzl) [inline src/hotspot/share/jvmci/jvmciCompilerToVM.cpp:553]: Unlike in the other cases, the indentation has not been updated. Looks good otherwise.

**Review by tschatzl — APPROVED** [review]

File: src/hotspot/share/jvmci/jvmciCompilerToVM.cpp (hunk 1 above)
Comment (kimbarrett) [reply src/hotspot/share/jvmci/jvmciCompilerToVM.cpp]: In other cases I was maintaining pre-existing alignment.  Here it was already formatted differently, and I
thought this was a file that used this indentation style (not that I like it).  But looking around, I'm not
finding other uses of this style, so I've aligned the arguments.

**Review by TheShermanTanker — APPROVED** [review]

**Comment by kimbarrett** [comment]
Thanks for reviews @tschatzl and @TheShermanTanker .

**Review by tschatzl — APPROVED** [review]

**Review by dougxc — APPROVED** [review]

**Comment by kimbarrett** [comment]
Thanks for reviews @tschatzl , @TheShermanTanker , and @dougxc

**Pull request closed without merging**